src/minichain/storage.py

```
@@ -0,0 +1,280 @@
+            """,
+            (head_hash,),
+        )
+
+    def load_chain_metadata(self) -> dict[str, int | str] | None:
+        """Load canonical chain metadata (height and head hash)."""
+        rows = self._connection.execute(
+            "SELECT key, value FROM chain_metadata WHERE key IN ('height', 'head_hash')"
+        ).fetchall()
+        if not rows:
+            return None
+        kv = {key: value for key, value in rows}
```

Ruff (0.15.2)

[convention] Unnecessary dict comprehension (rewrite using `dict()`)

Rewrite using `dict()`

(C416)